core/store/orm/orm.go

```diff
@@ -614,29 +614,33 @@ func (orm *ORM) AnyJobWithType(taskTypeName string) (bool, error) {
 	return found, ignoreRecordNotFound(rval)
 }
 
-// CreateTx returns a transaction by its surrogate key, if it exists, or
+// CreateTx finds and overwrites a transaction by its surrogate key, if it exists, or
 // creates it
 func (orm *ORM) CreateTx(tx *models.Tx) (*models.Tx, error) {
 	orm.MustEnsureAdvisoryLock()
 
 	err := orm.convenientTransaction(func(dbtx *gorm.DB) error {
 		var query *gorm.DB
+		foundTx := models.Tx{}
 		if tx.SurrogateID.Valid {
-			query = dbtx.First(&models.Tx{}, "surrogate_id = ?", tx.SurrogateID.ValueOrZero())
+			query = dbtx.First(&foundTx, "surrogate_id = ?", tx.SurrogateID.ValueOrZero())
+			// TODO: Put ON CONFLICT REPLACE ALL here
+			// https://github.com/sljeff/gorm/pull/5
 		} else {
-			query = dbtx.First(&models.Tx{}, "hash = ?", tx.Hash)
+			// TODO: Make hash unique so we can ON CONFLICT REPLACE ALL here
+			query = dbtx.First(&foundTx, "hash = ?", tx.Hash)
 		}
+		err := query.Error
 
-		ids := []uint64{}
-		err := query.Pluck("id", &ids).Error
 		if err != nil && err != gorm.ErrRecordNotFound {
 			return errors.Wrap(err, "CreateTx#First failed")
 		}
 
 		if err == gorm.ErrRecordNotFound {
 			return dbtx.Create(tx).Error
 		}
-		tx.ID = ids[0]
```

Comment:
I really don't know how this ever worked

Reply:
I believe there is a test for this, and it probably works because the DB is cleared out between tests. Perhaps if you add another record to that test to show how it could fail?

Reply:
Nope, it passed when I added another record too.

Reply:
I couldnt get it to fail.